master/buildbot/util/lineboundaries.py

```diff
@@ -20,21 +20,45 @@
 
 from twisted.internet import defer
 
+from buildbot.util.logger import Logger
+
+log = Logger()
 
-class LineBoundaryFinder(object):
 
-    __slots__ = ['partialLine', 'callback']
+class LineBoundaryFinder(object):
 
+    __slots__ = ['partialLine', 'callback', 'warned']
+    # split at reasonable line length.
+    # too big lines will fill master's memory, and slow down the UI too much.
+    MAX_LINELENGTH = 4096
     # the lookahead here (`(?=.)`) ensures that `\r` doesn't match at the end
     # of the buffer
-    newline_re = re.compile(r'(\r\n|\r(?=.)|\n)')
+    # we also convert cursor control sequence to newlines
```

Comment:
Does this mean that the control sequences will only be saved for "short" lines? If there's a process that output text with the control sequences that are then colored in the web UI, there would be nothing to color for the long lines?